### PR TITLE
fix: use install-tool for all

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Uses install-tool instead of install-pip, fixes some other related things

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/exec/containerbase.ts [hunk 1]
```diff
@@ -91,6 +103,11 @@ const allToolConfig: Record<string, ToolConfig> = {
     packageName: 'lerna',
     versioning: npmVersioningId,
   },
+  maven: {
+    datasource: 'maven',
+    packageName: 'org.apache.maven:maven',
```
Comment (rarkins) [inline lib/util/exec/containerbase.ts:108]: This is copied from https://github.com/containerbase/base/blob/ee496c11fa12e5e1b810544397071c9b8f906b4d/test/java/Dockerfile#L157-L158

Comment (viceice) [reply lib/util/exec/containerbase.ts:108]: we don't yet install maven. we only support the maven wrapper. so do we really need it now?

**Review by viceice — DISMISSED** [review]
otherwise LGTM

File: lib/util/exec/containerbase.ts (hunk 1 above)
Comment (rarkins) [reply lib/util/exec/containerbase.ts:108]: I'm reusing this for containerbase/runinstall. Maybe the list should be broken out into containerbase later instead?

Comment (viceice) [reply lib/util/exec/containerbase.ts:108]: ok, 🤔 so publish as new npm package to consume by renovate? or just provide as JSON file inside containerbase tools?

**Comment by viceice** [comment]
failed tests

File: lib/modules/manager/pipenv/__snapshots__/artifacts.spec.ts.snap [hunk 2]
```diff
@@ -150,7 +150,7 @@ exports[`modules/manager/pipenv/artifacts uses pipenv version from Pipfile 1`] =
     },
   },
   {
-    "cmd": "docker run --rm --name=renovate_sidecar --label=renovate_child -v "/tmp/github/some/repo":"/tmp/github/some/repo" -v "/tmp/renovate/cache":"/tmp/renovate/cache" -e PIPENV_CACHE_DIR -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w "/tmp/github/some/repo" containerbase/sidecar bash -l -c "install-tool python 3.10.2 && pip install --user pipenv==2020.8.13 && pipenv lock"",
+    "cmd": "docker run --rm --name=renovate_sidecar --label=renovate_child -v "/tmp/github/some/repo":"/tmp/github/some/repo" -v "/tmp/renovate/cache":"/tmp/renovate/cache" -e PIPENV_CACHE_DIR -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w "/tmp/github/some/repo" containerbase/sidecar bash -l -c "install-tool python 3.10.2 && install-tool pipenv ==2020.8.13 && pipenv lock"",
```
Comment (viceice) [inline lib/modules/manager/pipenv/__snapshots__/artifacts.spec.ts.snap:153]: this will cause issues, the `==` needs to be trimmed.

Comment (rarkins) [reply lib/modules/manager/pipenv/__snapshots__/artifacts.spec.ts.snap:153]: OK, I need to work out the best place to do this. Not sure why it's causing a problem only here

**Review by viceice — APPROVED** [review]

**Pull request merged**